src/Components/CommentAdder.jsx

```diff
@@ -0,0 +1,43 @@
+import { postComment } from "../api";
+import { useState } from "react";
+
+
+const CommentAdder = ({setComments, review_id}) => {
+    const [newComment, setNewComment] = useState("")
+    const [err, setError]=useState(null)
+
+    const handleSubmit = (e) => {
+        e.preventDefault()
+        const comment = {
+            "comment_id": Date.now(),
+            "body": newComment,
+            "review_id": review_id,
+            "author": "weegembump",
+            "votes": 0,
+            "created_at": "Just now",
+        }
+        setComments((currComments)=>{
+            return [comment, ...currComments]
+        })
+        postComment(review_id, newComment).then(()=>{
+            setNewComment("")
+        })
+        .catch((err)=>{
+            setError(true)
```

Comment:
If there is an error it would be good to remove the post you've optimisticaly rendered.